### PR TITLE
fix: mark flaky tests

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

It marks a bunch of tests as flaky based on the latest 21 builds.

## Why is it important?

We try to move the master branch to successful builds, we can not trust on a build that contains flaky and broken tests

## Checklist

- ~~[ ] My code follows the style guidelines of this project~~
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

closes https://github.com/elastic/beats/issues/21260

### CODE REVIEW

**Review by mdelapenya — APPROVED** [review]

File: auditbeat/module/file_integrity/eventreader_test.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ func init() {
 const ErrorSharingViolation syscall.Errno = 32
 
 func TestEventReader(t *testing.T) {
+	t.Skip("Flaky test: about 1/10 of bulds fails")
```
Comment (kuisathaverat) [inline auditbeat/module/file_integrity/eventreader_test.go:44]:
```suggestion
	t.Skip("Flaky test: about 1/10 of bulds fails https://github.com/elastic/beats/issues/21302")
```

**Comment by mdelapenya** [comment]
I'd like to bring to the table the importance to stop machines and fix the "flaky" tests. A test could be flaky for many reasons: CI slowness, CI setup, or even an undetected race condition.

I encourage the team to stop machines and debug the tests before sending new code. Otherwise the build will be broken for ever

File: auditbeat/module/file_integrity/eventreader_test.go [hunk 2]
```diff
@@ -240,6 +241,7 @@ func TestEventReader(t *testing.T) {
 }
 
 func TestRaces(t *testing.T) {
+	t.Skip("Flaky test: about 1/20 of bulds fails")
```
Comment (kuisathaverat) [inline auditbeat/module/file_integrity/eventreader_test.go:244]:
```suggestion
	t.Skip("Flaky test: about 1/20 of bulds fails https://github.com/elastic/beats/issues/21303")
```

**Review by v1v — APPROVED** [review]

File: filebeat/tests/system/test_autodiscover.py [hunk 3]
```diff
@@ -42,6 +45,8 @@ def test_docker(self):
         self.wait_until(lambda: self.log_contains('Stopping runner: input'))
         proc.check_kill_and_wait()
 
+    # 16/20 build fails
```
Comment (kuisathaverat) [inline filebeat/tests/system/test_autodiscover.py:48]:
```suggestion
    # 16/20 build fails https://github.com/elastic/beats/issues/21306
```

File: filebeat/tests/system/test_autodiscover.py [hunk 4]
```diff
@@ -11,6 +12,8 @@ class TestAutodiscover(filebeat.BaseTest):
     """
     Test filebeat autodiscover
     """
+    # 1/20 build fails
```
Comment (kuisathaverat) [inline filebeat/tests/system/test_autodiscover.py:15]:
```suggestion
    # 1/20 build fails https://github.com/elastic/beats/issues/21305
```

File: filebeat/tests/system/test_reload_inputs.py [hunk 5]
```diff
@@ -249,6 +250,8 @@ def test_load_configs(self):
         assert output[0]["message"] == first_line
         assert output[1]["message"] == second_line
 
+    # 1/20 build fails
```
Comment (kuisathaverat) [inline filebeat/tests/system/test_reload_inputs.py:253]:
```suggestion
    # 1/20 build fails https://github.com/elastic/beats/issues/21307
```

File: x-pack/auditbeat/tests/system/test_metricsets.py [hunk 6]
```diff
@@ -42,6 +43,8 @@ def test_metricset_login(self):
         # Metricset is beta and that generates a warning, TODO: remove later
         self.check_metricset("system", "login", COMMON_FIELDS + fields, config, warnings_allowed=True)
 
+    # 1/20 build fails
```
Comment (kuisathaverat) [inline x-pack/auditbeat/tests/system/test_metricsets.py:46]:
```suggestion
    # 1/20 build fails https://github.com/elastic/beats/issues/21308
```

File: filebeat/tests/system/test_autodiscover.py [hunk 7]
```diff
@@ -11,6 +12,8 @@ class TestAutodiscover(filebeat.BaseTest):
     """
     Test filebeat autodiscover
     """
+    # 1/20 build fails https://github.com/elastic/beats/issues/21305
+    @pytest.mark.flaky(reruns=1, reruns_delay=10)
```
Comment (jsoriano) [inline filebeat/tests/system/test_autodiscover.py:16]: Autodiscover tests should be fixed after https://github.com/elastic/beats/pull/21242

**Review by jsoriano — CHANGES_REQUESTED** [review]

File: libbeat/outputs/elasticsearch/client_integration_test.go [hunk 8]
```diff
@@ -54,7 +54,11 @@ func TestClientPublishEvent(t *testing.T) {
 	testPublishEvent(t, index, cfg)
 }
 
+// this test is broken 21 fails of 21 builds is not flaky at all
+// https://github.com/elastic/beats/issues/21295
+// https://github.com/elastic/beats/pull/21296
```
Comment (jsoriano) [inline libbeat/outputs/elasticsearch/client_integration_test.go:59]: Why not approving the original PR? :slightly_smiling_face: https://github.com/elastic/beats/pull/21296

Comment (kuisathaverat) [reply libbeat/outputs/elasticsearch/client_integration_test.go:59]: here is informative in order to have all the test, this change can be removed if we merge the other PR

File: libbeat/outputs/elasticsearch/client_integration_test.go [hunk 9]
```diff
@@ -54,7 +54,11 @@ func TestClientPublishEvent(t *testing.T) {
...
 func TestClientPublishEventKerberosAware(t *testing.T) {
+	t.Skip("Flaky test: https://github.com/elastic/beats/issues/21295")
```
Comment (kuisathaverat) [inline libbeat/outputs/elasticsearch/client_integration_test.go:61]:
```suggestion
func TestClientPublishEventKerberosAware(t *testing.T) {
```

File: filebeat/tests/system/test_autodiscover.py (hunk 7 above)
Comment (kuisathaverat) [inline filebeat/tests/system/test_autodiscover.py:16]:
```suggestion
```

File: filebeat/tests/system/test_autodiscover.py [hunk 10]
```diff
@@ -42,6 +45,8 @@ def test_docker(self):
         self.wait_until(lambda: self.log_contains('Stopping runner: input'))
         proc.check_kill_and_wait()
 
+    # 16/20 build fails https://github.com/elastic/beats/issues/21306
+    @pytest.mark.flaky(reruns=16, reruns_delay=10)
```
Comment (kuisathaverat) [inline filebeat/tests/system/test_autodiscover.py:49]:
```suggestion
```

File: filebeat/tests/system/test_autodiscover.py (hunk 7 above)
Comment (kuisathaverat) [reply filebeat/tests/system/test_autodiscover.py]: removed from the PR

File: filebeat/tests/system/test_autodiscover.py [hunk 11]
```diff
@@ -1,6 +1,7 @@
 import docker
 import filebeat
 import os
+import pytest
```
Comment (kuisathaverat) [inline filebeat/tests/system/test_autodiscover.py:4]:
```suggestion
```

**Review by jsoriano — APPROVED** [review]
Thanks!

**Comment by jsoriano** [comment]
@kuisathaverat could you please backport these changes to 7.x, 7.10 and 7.9 branches? They seem to be affected by tests skipped here https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats/detail/7.9/223/tests
Thanks!

**Pull request merged**